src/content/docs/develop/calling-rust.mdx

```diff
@@ -323,7 +480,56 @@ pub fn run() {
 }
 ```
 
-## Creating Multiple Commands
+### Accessing Raw Request
+
+Tauri commands can also access the full [`tauri::ipc::Request`] object which includes the raw body payload and the request headers.
+
+```rust
+#[derive(Debug, thiserror::Error)]
+enum Error {
+  #[error("unexpected request body")]
+  RequestBodyMustBeRaw,
+  #[error("missing `{0}` header")]
+  MissingHeader(&'static str),
+}
+
+impl serde::Serialize for Error {
+  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+  where
+    S: serde::ser::Serializer,
+  {
+    serializer.serialize_str(self.to_string().as_ref())
+  }
+}
+
+#[tauri::command]
+fn upload(request: tauri::ipc::Request) -> Result<(), Error> {
+  let tauri::ipc::InvokeBody::Raw(upload_data) = request.body() else {
+    return Err(Error::RequestBodyMustBeRaw);
+  };
+  let Some(authorization_header) = request.headers().get("Authorization") else {
+    return Err(Error::MissingHeader("Authorization"));
+  };
+
+  // upload...
+
+  Ok(())
+}
+```
+
+In the frontend you can call invoke() sending a raw request body by providing an ArrayBuffer or Uint8Array on the payload argument,
+and include request headers in the third argument:
+
+```js
+const data = new Uint8Array([1, 2, 3]);
+await __TAURI__.core.invoke('upload', data, {
+  headers: {
+    Authorization: 'apikey',
+  },
+});
+```
+
```

Comment:
We can add a note that  to enable raw request on Linux the feature `linux-ipc-protocol' must be used on the Rust side.

Reply:
Honestly with how flaky Linux support is i feel we should just raise the minimum webkitgtk version by one and get rid of the feature flag...

Reply:
how does that impact the already bad (?) list of supported distros again?

Reply:
Debain stable seems fine: https://tracker.debian.org/pkg/webkit2gtk
Fedora 40 seems to [deprecate it](https://fedoraproject.org/wiki/Changes/Remove_webkit2gtk-4.0_API_Version ) but still contains it by default in a newer version than required https://packages.fedoraproject.org/pkgs/webkit2gtk4.0/webkit2gtk4.0/ 
Ubuntu 22.04 is also fine (if you update it 😁 ) https://launchpad.net/ubuntu/jammy/+source/webkit2gtk
Other Arch based distros are gonna be fine anyway

Any other major distros I missed?
I think we can bump and remove 👍